core/src/main/java/org/infinispan/protostream/impl/ByteArrayOutputStreamEx.java

```diff
@@ -8,7 +8,9 @@
  *
  * @author anistor@redhat.com
  * @since 4.0
+ * @deprecated Use {@link LazyByteArrayOutputStream} instead.
  */
+@Deprecated(forRemoval = true)
```

Comment:
seeing this in the `impl` package, I'm tempted to say "remove it". 
Are we planning to backport this PR/optimizations for ProtoStream 5.x? We can deprecate it there and remove it here.

Reply:
> Are we planning to backport this PR/optimizations for ProtoStream 5.x? We can deprecate it there and remove it here.

I'm not sure. My gut feeling would be that it's not worth backporting, given that the main driver is the migration to protostream and the fact we only want new  features on `main`.

+1 to deprecating in `5.0.x`.